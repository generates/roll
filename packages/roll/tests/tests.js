import { test } from '@ianwalter/bff'
import { roll } from '../index.js'
import hello from './fixtures/hello.js'
import error from './fixtures/error.js'

test('Log info string to stdout', t => {
  const logger = roll.create({
    stdout: output => t.expect(output).toMatchSnapshot()
  })
  hello(logger)
})

test('Log error to stdout', t => {
  const logger = roll.create({
    stdout: output => t.expect(output).toMatchSnapshot()
  })
  error(logger)
})
