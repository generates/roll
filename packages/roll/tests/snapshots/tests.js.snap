// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Log error to stdout 1`] = `
"    Error: Woke up with my toolie what it do
    at default (file:///Users/ian/generates/roll/packages/roll/tests/fixtures/error.js:2:17)
    at file:///Users/ian/generates/roll/packages/roll/tests/tests.js:17:3
    at file:///Users/ian/generates/roll/node_modules/@ianwalter/bff/lib/runTest.js:26:26
    at new Promise (<anonymous>)
    at runTest (file:///Users/ian/generates/roll/node_modules/@ianwalter/bff/lib/runTest.js:24:7)
    at AsyncFunction.test (file:///Users/ian/generates/roll/node_modules/@ianwalter/bff/worker.js:137:15)
"
`;

exports[`Log info string to stdout 1`] = `
"    Hello World
"
`;
